### PR TITLE
Optimise QgsRectangle WKT methods

### PR DESCRIPTION
**Pull request by nyalldawson**

### CODE REVIEW

File: src/core/geometry/qgsrectangle.cpp [hunk 1]
```diff
@@ -112,16 +117,12 @@ QString QgsRectangle::asWktCoordinates() const
...
-    QStringLiteral( "))" );
-
-  return rep;
+  return QStringLiteral( "POLYGON((%1 %2, %3 %2, %3 %4, %1 %4, %1 %2))" ).arg(
+           qgsDoubleToString( mXmin ),
+           qgsDoubleToString( mYmin ),
+           qgsDoubleToString( mXmax ),
+           qgsDoubleToString( mYmax )
+         );
 }
```
Comment (strk) [inline src/core/geometry/qgsrectangle.cpp:126]: Shouldn't this method also consider Null rectangles and print as POLYGON EMPTY ?

File: tests/src/core/geometry/testqgsrectangle.cpp [hunk 2]
```diff
@@ -89,6 +89,36 @@ void TestQgsRectangle::fromWkt()
...
+  QVERIFY( ! rect.isEmpty() );
+  QCOMPARE( rect.xMinimum(), 0.0 );
+  QCOMPARE( rect.yMinimum(), 0.0 );
+  QCOMPARE( rect.xMaximum(), 1.0 );
+  QCOMPARE( rect.yMaximum(), 1.0 );
+  QVERIFY( rect == QgsRectangle::fromWkt( rect.asWktPolygon() ) );
+
+  QVERIFY( QgsRectangle::fromWkt( QStringLiteral( "xxx" ) ).isEmpty() );
+  QVERIFY( QgsRectangle::fromWkt( QStringLiteral( "LINESTRING ()" ) ).isEmpty() );
+  QVERIFY( QgsRectangle::fromWkt( QStringLiteral( "MULTIPOLYGON()" ) ).isEmpty() );
```
Comment (strk) [inline tests/src/core/geometry/testqgsrectangle.cpp:121]: I would check for .isNull() here, as .isEmpty() is also true for rectangles containing some information (ie: collapsed vertically and/or horizontally)

File: src/core/geometry/qgsrectangle.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/geometry/qgsrectangle.cpp:126]: I didn't want to overlap with your work -- I'll leave you to handle all the changes relating to null/empty

File: tests/src/core/geometry/testqgsrectangle.cpp (hunk 2 above)
Comment (nyalldawson) [reply tests/src/core/geometry/testqgsrectangle.cpp]: Done

**Pull request merged**